_posts/2023-09-13-prospero-add-feature-pack.adoc

```diff
@@ -0,0 +1,149 @@
+
+ ----
+# wildfly
+  manifest: file:/tmp/wildfly-29.0.1.Final/.installation/manifest-wildfly-29.0.1.Final.yaml
+  repositories:
+    id: central
+    url: https://repo1.maven.org/maven2/
+    id: jboss-public
+    url: https://repository.jboss.org/nexus/content/groups/public/
+    id: mrrc
+    url: https://maven.repository.redhat.com/ga/
+----
+
+== Creating feature pack channel
+
+All additional feature packs installed on top of Wildfly have to be registered in a channel. We're going to create a new manifest for this.
+
+[,bash]
+----
+echo "schemaVersion: 1.0.0
+streams:
+  - groupId: org.wildfly
+    artifactId: wildfly-datasources-galleon-pack
+    version: 5.0.0.Final
+" > /tmp/datasources-manifest.yaml
+----
+
+Then we can register Wildfly to receive updates from a new channel:
+
+[,bash]
+----
+./prospero channel add \
+ --channel-name datasources \
+ --manifest datasources-manifest.yaml \
+ --repositories central::https://repo1.maven.org/maven2 \
+ --dir /tmp/wildfly
+ Subscribing /tmp/wildfly to channel datasources
+----
+
+
+== Install wildfly-datasource-galleon-pack
+
+Finally to install the MariaDB database driver from wildfly-datasource-galleon-pack:
+
+[,bash]
+----
+./prospero.sh feature-pack add \
+  --fpl org.wildfly:wildfly-datasources-galleon-pack \
+  --layers mariadb-datasource \
+  --dir /tmp/wildfly
+Installing feature pack `org.wildfly:wildfly-datasources-galleon-pack` in server `/tmp/wildfly`
+Continue adding the feature pack? [y/N]:y
+Applying changes
+
+Feature-packs resolved.
+Packages installed.
+Downloaded artifacts.
+JBoss modules installed.
+Configurations generated.
+JBoss examples installed.
+Operation completed in 19.62 seconds.
+----
+
+This command will create a new module `org.mariadb.jdbc:main` and will update `standalone/configuration/standalone.xml` 
+with a new driver and datasource configuration.
+
+== Updating mariadb driver version
+
+Steps above installed a default version of MariaDB driver, but using prospero and wildfly channels we are able to udate that.
+
+First, lets add desired version of the driver to the datasources manifest:
+
+[,bash]
+----
+echo "
+  - groupId: org.mariadb.jdbc
+    artifactId: mariadb-java-client
+    version: 3.2.0
+" >> /tmp/datasources-manifest.yaml
+----
+
+Next, we just need to let prospero perform an update:
+
+[,bash]
+----
+./prospero.sh update perform --dir /tmp/wildfly
+Updates found:
+  org.mariadb.jdbc:mariadb-java-client                  3.1.4                ==>  3.2.0
+Continue with update [y/N]: y
+Building updates
+
+Feature-packs resolved.
+Packages installed.
+Downloaded artifacts.
+JBoss modules installed.
+Configurations generated.
+JBoss examples installed.
+
+Build update complete!
+Applying updates
+Update complete!
+Operation completed in 25.48 seconds.
+----
+
+The `org.mariadb.jdbc:main` has been updated with new driver version:
+
+[,bash]
+----
+ls /tmp/wildfly/modules/org/mariadb/jdbc/main
+mariadb-java-client-3.2.0.jar  module.xml
+----
```

Comment:
Is it interesting to see a snippet of the output of `/tmp/wildfly/bin/standalone.sh`?

Could you please add a Conclusion that summarise what we learnt from this article.